src/snowflake/snowpark/mock/_udf.py

```diff
@@ -114,37 +114,27 @@ def _do_register_udf(
         _emit_ast: bool = True,
         **kwargs,
     ) -> UserDefinedFunction:
+        if kwargs.get("_registered_object_name") is not None:
```

Comment:
Can we define ast, ast_id first as variables? This may else lead to an unbound local error issue.